02_C# Fundamentals/SystemWatcherApp/SystemWatcherApp/Program.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using SystemWatcherApp.Configs;
+using SystemWatcherApp.Resources;
+
+namespace SystemWatcherApp
+{
+    class Program
+    {
+        readonly static List<string> startDirectories;
+        readonly static List<FileSystemWatcher> watchers;
+        readonly static string defaultDirectory = @"D:\_02_BCL\Default\";
+        static CultureInfo currentCulture;
+        static List<RuleElement> rules;
+
+        static Program()
+        {
+            rules = new List<RuleElement>();
+            startDirectories = new List<string>();
+            watchers = new List<FileSystemWatcher>();
+        }
+
+        static void Main(string[] args)
+        {
+            Console.OutputEncoding = Encoding.UTF8;
+
+            WatcherSectionGroup watcherSectionGroup = ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None).SectionGroups["watcherSection"] as WatcherSectionGroup;
```

Comment:
I never seen method OpenExeConfiguration, why is this needed? Can't we just do `ConfigurationManager.Section["abc123"]`?

Reply:
No, we cannot, because for `ConfigurationSectionGroup` the simpler approach does not work. They suggest using `OpenExeConfiguration` https://docs.microsoft.com/en-us/dotnet/api/system.configuration.configurationsectiongroup?view=netframework-4.8#examples. This is why I have chosen this way

Reply:
Aaaah you are using SectionGroup... Why? Can we just use ConfigurationSection? Look at how Marina is using it, it looks very simple:
![image](https://user-images.githubusercontent.com/51448679/76302273-b9507980-62d0-11ea-9172-2f7ff007f0e9.png)

Reply:
Yes, I can. I have just decided to collect sections in the separate group. Should I give up the `ConfigurationSectionGroup`?

Reply:
I think we can leave it as is, as you already implemented it, but I think that was an unnecessary complication.